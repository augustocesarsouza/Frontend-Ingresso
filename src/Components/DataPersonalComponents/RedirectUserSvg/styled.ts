import styled from 'styled-components';

interface ContainerSvgSpanProps {
  $svgactive: string;
}

export const ContainerSvgSpan = styled.div<ContainerSvgSpanProps>`
  width: 100%;
  display: flex;
  align-items: center;
  padding: 10px;
  background: #80808021;
  border-left: 5px solid #039de2;
  cursor: pointer;
  background: ${props => props.$svgactive === "true" && "rgb(255, 115, 0)"};
  border-left: ${props => props.$svgactive === "true" && "none"};
  gap: 8px;
`

interface ContainerSvgProps {
  $svgactive: string;
}

export const ContainerSvg = styled.div<ContainerSvgProps>`

  svg {
    fill: #2196F3;
    display: block;
    min-height: 33px;
    fill: ${props => props.$svgactive === "true" && "#ffffff"};
  }
`

interface SpanProps {
  $svgactive: string;
}

export const Span = styled.span<SpanProps>`
  color: #2196F3;
  font-size: 15px;
  text-align: center;
  display: block;
  color: ${props => props.$svgactive === "true" && "#ffffff"};
`