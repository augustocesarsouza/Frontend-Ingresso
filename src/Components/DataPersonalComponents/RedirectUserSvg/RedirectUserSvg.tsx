import * as Styled from './styled';

const RedirectUserSvg = ({ svg, text, svgActive }) => {
  return (
    <Styled.ContainerSvgSpan $svgactive={String(svgActive)}>
      <Styled.ContainerSvg $svgactive={String(svgActive)}>{svg}</Styled.ContainerSvg>
      <Styled.Span $svgactive={String(svgActive)}>{text}</Styled.Span>
    </Styled.ContainerSvgSpan>
  );
};

export default RedirectUserSvg;
