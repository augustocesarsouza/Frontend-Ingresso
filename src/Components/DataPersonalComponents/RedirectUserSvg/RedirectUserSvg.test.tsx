import { render, screen } from '@testing-library/react';
import RedirectUserSvg from './RedirectUserSvg';
import TicketSvg from '../../../Svg/TicketSvg';

describe('RedirectUserSvg', () => {
  test('should render', () => {
    const active = false;
    const { container } = render(
      <RedirectUserSvg svg={<TicketSvg />} text={'Meus Pedidos'} svgActive={active} />
    );

    const svgTicked = container.querySelector('.ticket-svg');
    expect(svgTicked).toBeInTheDocument();

    const spanMyOrders = screen.getByText('Meus Pedidos');
    expect(spanMyOrders).toBeInTheDocument();

    expect(container).toMatchSnapshot();

    expect.assertions(3);
  });

  test('should render', () => {
    const active = true;
    const { container } = render(
      <RedirectUserSvg svg={<TicketSvg />} text={'Meus Pedidos'} svgActive={active} />
    );

    expect(container).toMatchSnapshot();
  });
});
