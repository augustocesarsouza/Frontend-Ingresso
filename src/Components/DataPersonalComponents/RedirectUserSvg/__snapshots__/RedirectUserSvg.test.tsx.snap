// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedirectUserSvg should render 1`] = `
.c0 {
  width: 100%;
  display: flex;
  align-items: center;
  padding: 10px;
  background: #80808021;
  border-left: 5px solid #039de2;
  cursor: pointer;
  gap: 8px;
}

.c1 svg {
  fill: #2196F3;
  display: block;
  min-height: 33px;
}

.c2 {
  color: #2196F3;
  font-size: 15px;
  text-align: center;
  display: block;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        class="ticket-svg"
        height="30"
        role="img"
        viewBox="0 0 32 33"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.775 24.309c.19.329.606.446.926.261l9.93-5.733.692 1.197-9.931 5.734c-.964.557-2.213.206-2.785-.785l-.719-1.245s1.564-1.382.415-3.371c-1.149-1.99-2.97-1.054-2.97-1.054l-.877-1.52c-.572-.99-.252-2.245.714-2.802l9.93-5.734.692 1.197-9.931 5.734a.678.678 0 0 0-.237.93l.354.613c.132-.01.27-.015.413-.011 1.218.03 2.329.672 3.08 1.972.73 1.264.776 2.538.308 3.663-.078.188-.165.36-.256.517l.252.437ZM21.225 8.69a.681.681 0 0 0-.926-.261l-9.93 5.733-.692-1.197 9.931-5.734c.964-.557 2.213-.206 2.785.785l.719 1.245s-1.564 1.382-.415 3.371c1.149 1.99 2.97 1.054 2.97 1.054l.877 1.52c.572.99.252 2.245-.714 2.802l-9.93 5.734-.692-1.197 9.931-5.734a.678.678 0 0 0 .237-.93l-.354-.613c-.132.01-.27.015-.413.011-1.218-.03-2.329-.672-3.08-1.972-.73-1.264-.776-2.538-.308-3.663.078-.188.165-.36.256-.517l-.252-.437Z"
        />
      </svg>
    </div>
    <span
      class="c2"
    >
      Meus Pedidos
    </span>
  </div>
</div>
`;

exports[`RedirectUserSvg should render 2`] = `
.c0 {
  width: 100%;
  display: flex;
  align-items: center;
  padding: 10px;
  background: #80808021;
  border-left: 5px solid #039de2;
  cursor: pointer;
  background: rgb(255, 115, 0);
  border-left: none;
  gap: 8px;
}

.c1 svg {
  fill: #2196F3;
  display: block;
  min-height: 33px;
  fill: #ffffff;
}

.c2 {
  color: #2196F3;
  font-size: 15px;
  text-align: center;
  display: block;
  color: #ffffff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        class="ticket-svg"
        height="30"
        role="img"
        viewBox="0 0 32 33"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.775 24.309c.19.329.606.446.926.261l9.93-5.733.692 1.197-9.931 5.734c-.964.557-2.213.206-2.785-.785l-.719-1.245s1.564-1.382.415-3.371c-1.149-1.99-2.97-1.054-2.97-1.054l-.877-1.52c-.572-.99-.252-2.245.714-2.802l9.93-5.734.692 1.197-9.931 5.734a.678.678 0 0 0-.237.93l.354.613c.132-.01.27-.015.413-.011 1.218.03 2.329.672 3.08 1.972.73 1.264.776 2.538.308 3.663-.078.188-.165.36-.256.517l.252.437ZM21.225 8.69a.681.681 0 0 0-.926-.261l-9.93 5.733-.692-1.197 9.931-5.734c.964-.557 2.213-.206 2.785.785l.719 1.245s-1.564 1.382-.415 3.371c1.149 1.99 2.97 1.054 2.97 1.054l.877 1.52c.572.99.252 2.245-.714 2.802l-9.93 5.734-.692-1.197 9.931-5.734a.678.678 0 0 0 .237-.93l-.354-.613c-.132.01-.27.015-.413.011-1.218-.03-2.329-.672-3.08-1.972-.73-1.264-.776-2.538-.308-3.663.078-.188.165-.36.256-.517l-.252-.437Z"
        />
      </svg>
    </div>
    <span
      class="c2"
    >
      Meus Pedidos
    </span>
  </div>
</div>
`;
